client/src/redux/donation-saga.js

```diff
@@ -22,6 +37,40 @@ function* showDonateModalSaga() {
   }
 }
 
+function* addDonationSaga({ payload }) {
+  try {
+    yield call(addDonation, payload);
+    yield put(addDonationComplete());
+  } catch (error) {
+    const data =
+      error.response && error.response.data
+        ? error.response.data
+        : {
+            message: defaultDonationError
+          };
+    yield put(addDonationError(data.message));
+  }
+}
+
+function* postChargeStripeSaga({ payload }) {
+  try {
+    yield call(postChargeStripe, payload);
+    yield put(postChargeStripeComplete());
+  } catch (error) {
+    const data =
+      error.response && error.response.data
+        ? error.response.data
+        : {
+            error: defaultDonationError
+          };
+    yield put(postChargeStripeError(data.error));
```

Comment:
```suggestion
    const err =
      error.response && error.response.data
        ? error.response.data.error
        : defaultDonationError
    yield put(postChargeStripeError(err));
```
I think this is a little cleaner.